substrate/frame/support/src/lib.rs

```diff
@@ -2262,6 +2262,221 @@ pub mod pallet_macros {
+	///
+	/// There are three types of queries:
+	///
+	/// 1. [`OptionQuery`](frame_support::storage::types::OptionQuery): The default query type. It returns `Some(V)` if the value is present, or 
+	/// `None` if it isn't, where `V` is the value type.
+	/// 2. [`ValueQuery`](frame_support::storage::types::ValueQuery): Returns the value itself if present; otherwise, it returns 
+	/// `Default::default()`. This behavior can be adjusted with the `OnEmpty` generic parameter, which defaults to `OnEmpty = GetDefault`.
+	/// 3. [`ResultQuery`](frame_support::storage::types::ResultQuery): Returns `Result<V, E>`, where `V` is the value type, and `E` is the pallet error type defined by the [`#[pallet::error]`](frame_support::pallet_macros::error) attribute.
+	///
+	/// ### Appending
+	///
+	/// All storage items — such as `StorageValue`, `StorageMap`, and their variants—offer an `::append()` method 
+	/// optimized for collections. Using this method avoids the inefficiency of decoding and re-encoding 
+	/// entire collections when adding items. For instance, consider the storage declaration 
+	/// `type MyVal<T> = StorageValue<_, Vec<u8>, ValueQuery>`. With `MyVal` storing a large list of bytes, 
+	/// `::append()` lets you directly add bytes to the end in storage without processing the full list. 
+	/// Depending on the storage type, additional key specifications may be needed.
+	///
+	/// ### Optimized Length Decoding
+	///
+	/// All storage items — such as `StorageValue`, `StorageMap`, and their counterparts — 
+	/// incorporate the `::decode_len()` method. This method allows for efficient retrieval 
+	/// of a collection's length without the necessity of decoding the entire dataset.
+	///
+	/// ### Hashers
+	///
+	/// For all storage types, except `StorageValue`, a set of hashers needs to be specified. The choice
+	/// of hashers is crucial, especially in production chains. The purpose of
+	/// storage hashers in maps is to ensure the keys of a map are uniformly distributed. An
+	/// unbalanced map/trie can lead to inefficient performance.
+	///
+	/// In general, hashers are categorized as either cryptographically secure or not. The former is slower than the
+	/// latter. `Blake2` and `Twox` serve as examples of each, respectively.
+	///
+	/// As a rule of thumb:
+	///
+	/// 1. If the map keys are not controlled by end users, or are cryptographically secure by
+	/// definition (e.g., `AccountId`), then the use of cryptographically secure hashers is NOT required.
+	/// 2. If the map keys are controllable by the end users, cryptographically secure hashers should
+	/// be used.
+	///
+	/// For more information, look at the types that implement
+	/// [`frame_support::StorageHasher`](frame_support::StorageHasher).
+	///
+	/// Lastly, it's recommended for hashers with "concat" to have reversible hashes. Refer to the
+	/// implementors section of [`hash::ReversibleStorageHasher`](frame_support::hash::ReversibleStorageHasher).
+	///
+	/// ### Prefixes
```

Comment:
ahh, so maybe you don't need to cover the key generation details of each storage type in each type, and tackle them all here?

Reply:
I have given an example for StorageValue and StorageMap. However, I'm still thinking about whether I should add them additionally for each type in their respective pages...